lib/mix/lib/mix/tasks/format.ex

```diff
@@ -360,10 +360,21 @@ defmodule Mix.Tasks.Format do
     map =
       for input <- List.wrap(formatter_opts[:inputs]),
           file <- Path.wildcard(Path.join(prefix ++ [input]), match_dot: true),
-          do: {file, formatter_opts},
+          do: {file, {dot_formatter, formatter_opts}},
           into: %{}
 
-    Enum.reduce(subs, Map.merge(acc, map), fn {sub, formatter_opts_and_subs}, acc ->
+    acc =
+      Map.merge(acc, map, fn file, {dot_formatter1, _}, {dot_formatter2, formatter_opts} ->
+        Mix.shell().error(
+          "Both #{dot_formatter1} and #{dot_formatter2} specify the file #{file} " <>
+            "in their :inputs option. To resolve the conflict, the configuration in " <>
+            "#{dot_formatter1} will be ignored"
```

Comment:
If we are actually resolving the conflicts, then I would rather document how we solve it and get rid of the warning altogether. For instance, I would say the one closest to the subdirectory always wins. This means the inputs files are simpler. Thoughts?

Reply:
There are some edge cases in which telling which `.formatter.exs` is closer to a file cannot be accurate. They are quite edgy though.

For example:

```
.formatter.exs # [inputs: "lib/**/*.ex", subdirectories: ["lib/bar", "lib/foo"]]
lib/bar/.formatter.exs # [inputs: ["../*.ex", "/*.ex"]]
lib/foo/.formatter.exs # [inputs: "../*.ex"]
lib/a.ex
```

Btw, the example above was not printing errors. I just pushed some changes to deal with `..` and `.` in the paths.

Reply:
I see, the wording would have to be the innermost subdirectory that matches it. Although I guess we can start with a warning for now and we can always amend it based on feedback!